lib/_http_incoming.js

```diff
@@ -144,6 +146,22 @@ ObjectDefineProperty(IncomingMessage.prototype, 'trailers', {
   }
 });
 
+ObjectDefineProperty(IncomingMessage.prototype, 'signal', {
+  get: function() {
+    if (!this[kAbortController]) {
+      this[kAbortController] = new AbortController();
```

Comment:
I think accessing a property should not have any side effect.

Reply:
I think the side effect is not observable outside of the property though so should be fine? Something about if a tree falls in a forest and no one hears.

Would you prefer an implementation where every HTTP request is larger and slower because of the `AbortSignal`?

Reply:
I'm fine with the lazy creation of the internal state. Calling the `abort()` inside the getter may be very surprising for folks but there's not another way of creating an `AbortController` that is already tripped so I'm not sure there's a good approach for that.

Reply:
As I said, my only concern is on calling the `abort()` and that's only because I have a general dislike for getters that have potentially unexpected side effects. In this case, I think it's fine.

Reply:
> Would you prefer an implementation where every HTTP request is larger and slower because of the AbortSignal?

That's the exact thing I want to prevent from happening.

Calling this `.signal()` instead of `.signal` is a clear indication that calling this method has side effects.

Reply:
I agree but from the user's point of view it has 0 side effects they are all internal :)